python/ray/tune/tune.py

```diff
@@ -1165,12 +1165,20 @@ class and registered trainables.
             )
 
     if _use_storage_context():
+        # TODO(ekl) construct the storage context just once at the start of the run.
+        # We need it here to resolve the filesystem relative to the given path.
+        storage = StorageContext(
+            storage_path=storage_path,
+            storage_filesystem=storage_filesystem,
+            sync_config=sync_config,
+            experiment_dir_name=name,
+        )
```

Comment:
Ah, good catch. I'm not testing the custom fs in the experiment analysis suite yet.

Can we actually pull this from `experiments[0].storage.storage_filesystem` instead of recreating the storage context?